crates/subspace-farmer/src/plot.rs

```diff
@@ -385,6 +390,46 @@ impl Plot {
         })?
     }
 
+    /// Returns sequential piece indexes for piece retrieval
```

Comment:
Why do you need piece indexes? I would have expected you need pieces as well

Reply:
Updated description of the pr with the algorithm of that iterator.